### PR TITLE
Added python interface for multiple start indices

### PR DESCRIPTION
**Pull request by PhilippDahlinger**
- Correct typing
- assert statement to check that start list <= num_samples
- conversion to unsigned int np array

Added rust wrappers to handle multiple start indices
- New start_idx enum to handle both cases
- Added Rust input check
- Rust interface setup for implementing multi start index algorithms.

### CODE REVIEW

**Review by leonardodalinky — CHANGES_REQUESTED** [review]
Do you try to compile the code and test if the NotImplementError occurs?

I think the `StartIndex` enum should derive the `#[derive(FromPyObject)]` features as described in [pyo3-doc](https://pyo3.rs/v0.22.5/conversions/traits) and [an example here](https://users.rust-lang.org/t/closest-implementation-for-enums-with-fields-in-pyo3/89385/7), if the `StartIndex` wrapper seems to be more fancy.

Though, I may choose to simply change the `start_idx` to a single list in the Python interface to avoid the wrapper and complex if-else/match conditions in Rust.

Anyways, both way is OK to me.

**Comment by PhilippDahlinger** [comment]
I didn't want to make any interface changes to the existing code, so I chose the enum. I compiled the code and the NotImplementedError occurs (after fixing some rust stuff), when provided the start index list. The tests still pass when using no start idx or a single start index.

File: bench/test_bench.py
Comment (leonardodalinky) [inline bench/test_bench.py:1]: Function name `test_vanilla_fps_4k`  is already in use. Try another name like `test_vanilla_fps_4k_multiple`.

**Review by leonardodalinky — CHANGES_REQUESTED** [review]
Seems good to me with only minor errors. Do you want to support multiple start indices to all functions here?

File: bench/test_bench.py
Comment (PhilippDahlinger) [reply bench/test_bench.py:1]: Thanks, I changed it.

**Comment by PhilippDahlinger** [comment]
Thanks! Well, its my first contact with Rust, but it looks like a nice language, I might want to learn it. Ideally, It would be nice to have multiple start indices for all fps functions, but if you could only start with `KDTree-based FPS` or `Bucket-based FPS`, I would be already completely happy, since then I can use it already in my current work. For my problem, both work very well and are really fast.

**Comment by leonardodalinky** [comment]
OK，I would try to make these features at this weekend. :)

**Review by leonardodalinky — COMMENTED** [review]
LGTM.

**Pull request merged**